### PR TITLE
feat: complete issue #1 training pipeline scaffolding

### LINKED ISSUE #1: Roadmap: DogYOLOv11 데이터 파이프라인 및 학습 준비
## 배경
현재 레포는 DogYOLOv11 모델 구조, JSON 어노테이션 계약(body/head/emotion/action), loss 정합, 디코더/타깃 변환 유틸이 반영된 상태입니다.

## 목표
인스타 기반 강아지 데이터 수집/정제 이후 바로 학습 가능한 파이프라인으로 확장합니다.

## TODO
- [ ] `torch` 환경에서 `smoke_test.py` 실행/검증
- [ ] JSON 데이터셋 로더(`Dataset`/`DataLoader`) 추가
- [ ] 학습 스크립트(`train.py`) 추가
- [ ] 검증/평가 스크립트(`val.py`) 추가
- [ ] 추론 후처리(NMS) 추가
- [ ] 라벨 검증기(박스 범위/Unknown 정책 검사) 추가
- [ ] README에 실행 순서 및 예시 업데이트

## 완료 기준
- 단일 커맨드로 smoke test 통과
- 샘플 데이터로 1 epoch 학습 가능
- 추론 결과가 JSON 계약 포맷으로 저장됨

### PR DESCRIPTION
**Pull request by stupidcoderJung**
## Summary
- add `dataset.py` with manifest-based `DogJsonDataset` and `dog_collate_fn`
- add `label_validator.py` CLI for bbox/label validation
- add `train.py` with end-to-end training loop and synthetic sample generation (`--synthetic-samples`)
- add `val.py` for eval/inference JSON export and NMS before/after stats
- update `smoke_test.py` to log NMS before/after counts
- update README with beginner-friendly data->validate->train->val workflow
- extend CI smoke workflow to compile new scripts and run train/val pipeline smoke
- refresh ADR tradeoff/follow-up section for current NMS status

## Test
- `source .venv/bin/activate && python -m py_compile dog_yolov11.py dataset.py label_validator.py train.py val.py smoke_test.py`
- `source .venv/bin/activate && python smoke_test.py --img-h 320 --img-w 320 --batch 1 --num-breeds 120 --num-emotions 5 --num-actions 5 --width-mult 0.23 --device cpu`
- `source .venv/bin/activate && python train.py --synthetic-samples 4 --epochs 1 --batch-size 2 --img-h 320 --img-w 320 --width-mult 0.23 --device cpu --max-steps-per-epoch 1 --run-name issue1-final`
- `source .venv/bin/activate && python label_validator.py --manifest runs/train/issue1-final-<timestamp>/synthetic_dataset/manifest.json --allow-unknown-breed --max-messages 5`
- `source .venv/bin/activate && python val.py --manifest runs/train/issue1-final-<timestamp>/synthetic_dataset/manifest.json --checkpoint runs/train/issue1-final-<timestamp>/last.pt --img-h 320 --img-w 320 --width-mult 0.23 --batch-size 2 --device cpu --max-batches 1 --output-json runs/train/issue1-final-<timestamp>/val_predictions.json`

Closes #1

### CODE REVIEW

**Comment by stupidcoderJung** [comment]
추가 Codex 리뷰 2건 반영 완료했습니다. 방금 커밋 `86960a9` 푸시했습니다.

반영 내용:
- `val.py`
  - 체크포인트에서 `breed_names`를 로드한 경우에는 `Unknown` 클래스를 자동 append하지 않도록 수정
  - `--unknown-breed-policy class` + 기존 체크포인트 조합에서 strict 로딩 shape mismatch가 나지 않도록 보완
- `README.md`
  - 설치 섹션의 로컬 절대경로(`cd /Users/...`)를 제거하고 `git clone` + `cd dog-yolov11-k9-json`로 수정

검증:
- `train.py --synthetic-samples ...`로 생성한 체크포인트에 대해
- `val.py --unknown-breed-policy class` 실행 시 정상 동작 확인

**Comment by stupidcoderJung** [comment]
@codex review
